src/core/platformutilities.h

```diff
@@ -47,7 +47,7 @@ class PlatformUtilities : public QObject
      * @param prefix The folder where the picture should be put
      * @return The name of the picture or null
      */
-    Q_INVOKABLE virtual PictureSource *getPicture( const QString &prefix );
+    Q_INVOKABLE virtual PictureSource *getPicture( const QString &prefix, const QString &source );
```

Comment:
Could the source string be replaced with an enum?

We can still leave it in place for the java api if that's complicated to do with JNI. I'd just like to avoid having to use those magic (and undocumented ;) ) strings in every native provider.

Reply:
I don't know. 
The `source` string is set in the `ExternalResources.qml` when I call the `platformUtilities.getPicture` function to define whether I want the picture from the camera or from the gallery and then is stored as parameter in the Intent and read from the Java code.

So the enum should be usable from qml and java. I don't know if it's possible.
 
Another possibility can be to define different functions in `platformutilities.cpp` for camera and gallery and also define different intents in the java part but this would lead to duplicating several portions of the code.

Reply:
It should be possible: https://doc.qt.io/qt-5/qml-enumeration.html
The alternative of two different function also sounds good (to reduce code duplication, they could use an internal helper function). Anything that avoids "magic strings" will be good :)